### ISSUE TITLE
radio-button: unable to select a radio button using space key

### ISSUE THREAD

**Issue by kittenish** [angular/material#11960]
## Bug

# Demo and steps to reproduce the issue
[This blank StackBlitz demo](https://stackblitz.com/edit/angularjs-material-blank?file=app%2Fapp.template.html) can be used to create a reproduction that demonstrates your issue.

**Demo URL (required)***:  https://codepen.io/jiaruigao/pen/VweGJrp?editable=true%3Dhttps%3A%2F%2Fmaterial.angularjs.org%2FHEAD%2Fdemo%2FradioButton

#### Detailed Reproduction Steps
1. Do not pre-select any radio button within a radio group
2. Navigate to a radio button using keyboard tab key and then press space key
3. Observe the radio button is not selected.

### Explain the expected behavior
The radio button is expected to be selected in step 3.

### Explain the current behavior
The radio button is not selected.

### Discuss the use-case or motivation for changing the existing behavior
https://www.w3.org/TR/wai-aria-practices-1.1/#radiobutton
In the keyboard interaction section, the user should be to check the focused radio button if it is not already checked using space key.

### List the affected versions of AngularJS, Material, OS, and browsers

- AngularJS: 1.
- AngularJS Material: 1.
- OS: Mac OS Version 10.15.3
- Browsers: Chrome 81.0.4044.113 
 
It could be reproduced by removing the pre-selection on the [doc page](https://material.angularjs.org/HEAD/demo/radioButton).

https://codepen.io/jiaruigao/pen/VweGJrp?editable=true%3Dhttps%3A%2F%2Fmaterial.angularjs.org%2FHEAD%2Fdemo%2FradioButton

#### Stack Trace
```

```

**Comment by Splaktar** [comment]
Thank you for this report. I've confirmed that this is an issue that needs to be resolved.

In https://github.com/angular/material/commit/03c7592798f904ac7a59b4a1c580672ca7c4789f, we added support for the `ENTER` key submitting a form and we already had up/down/left/right arrow key behaviors, but we don't have any behavior for the `SPACE` bar key.

That said, moving the selection using the up/down/left/right keys does select the newly focused radio button option as defined in the ARIA practices. Also focusing the radio group does appear to follow the ARIA practices.

We need to select the focused radio button if it is not already selected on `SPACE`.

**Comment by kittenish** [comment]
Thanks for addressing the issue. Yes we see radio items are selectable via the arrow keys, it would be great if we could also select via `SPACE` key.
